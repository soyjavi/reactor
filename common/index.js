import { LAYOUT } from './layout/index';
import SHAPE from './shape';
import THEME from './theme';

export { LAYOUT, SHAPE, THEME };
