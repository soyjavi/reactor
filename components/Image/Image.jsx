import { func } from 'prop-types';
import React, { useState } from 'react';
import { Image as ImageNative, View } from 'react-native';

import Skeleton from '../Skeleton';
import styles from './Image.style';

const Image = ({ onLoad, ...inherit }) => {
  const [ready, setReady] = useState(false);

  return (
    <View style={[styles.container, inherit.styleContainer || inherit.style]}>
      <ImageNative
        {...inherit}
        onLoad={(event) => {
          setReady(true);
          onLoad(event);
        }}
      />
      {!ready && <Skeleton style={styles.skeleton} />}
    </View>
  );
};

Image.propTypes = {
  onLoad: func,
};

Image.defaultProps = {
  onLoad() {},
};

export default Image;
