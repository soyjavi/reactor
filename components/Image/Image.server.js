import { createElement } from 'react';

import { serverSideStyle } from '../../common';

export default ({ alt = 'reactor', source: { uri }, style, title }) =>
  createElement('img', {
    alt,
    src: uri,
    style: serverSideStyle(style),
    title,
  });
