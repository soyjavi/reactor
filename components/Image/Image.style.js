import { StyleSheet } from 'react-native';

import { LAYOUT } from '../../common';

export default StyleSheet.create({
  container: {
    ...LAYOUT.STYLE.CENTERED,
  },

  skeleton: {
    position: 'absolute',
    left: 0,
    top: 0,
    width: '100%',
    height: '100%',
  },
});
