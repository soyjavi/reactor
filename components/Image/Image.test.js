import React from 'react';
import renderer from 'react-test-renderer';

import { THEME } from '../../common';
import Image from './Image';

const dataSource = [
  { caption: 'Spain', href: '/' },
  { caption: 'Basque Country', href: '/' },
  { caption: 'Bilbao', href: '/' },
];

const IMAGE_PLACEHOLDER = 'https://picsum.photos/320/200/?random';

jest.useFakeTimers();

describe('<Image>', () => {
  it('renders', () => {
    const tree = renderer.create(<Image />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {source}', () => {
    const tree = renderer.create(<Image source={{ uri: IMAGE_PLACEHOLDER }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {color} (only Native)', () => {
    const tree = renderer.create(<Image color={THEME.COLOR.ERROR} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {invert} (only Web)', () => {
    const tree = renderer.create(<Image invert />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {style}', () => {
    const tree = renderer.create(<Image style={{ backgroundColor: THEME.COLOR.ERROR }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
