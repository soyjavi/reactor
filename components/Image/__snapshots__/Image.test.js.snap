// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image> renders 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Image
    onLoad={[Function]}
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#e3e3e3",
          "minHeight": 8,
          "minWidth": 8,
        },
        Array [
          undefined,
          Object {
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          },
        ],
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;

exports[`<Image> when {source} 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Image
    onLoad={[Function]}
    source={
      Object {
        "uri": "https://picsum.photos/320/200/?random",
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#e3e3e3",
          "minHeight": 8,
          "minWidth": 8,
        },
        Array [
          undefined,
          Object {
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          },
        ],
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": "100%",
          "opacity": 0,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;

exports[`<Image> when {style} 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "#e85a7e",
      },
    ]
  }
>
  <Image
    onLoad={[Function]}
    style={
      Object {
        "backgroundColor": "#e85a7e",
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#e3e3e3",
          "minHeight": 8,
          "minWidth": 8,
        },
        Array [
          Object {
            "backgroundColor": "#e85a7e",
          },
          Object {
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          },
        ],
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;
